src/roswire/definitions/package.py

```diff
@@ -100,18 +100,29 @@ def paths(shell: dockerblade.Shell,
               files: dockerblade.FileSystem
               ) -> List[str]:
         """Parses :code:`ROS_PACKAGE_PATH` for a given shell."""
-        path_str = shell.environ('ROS_PACKAGE_PATH')
-        package_paths: List[str] = path_str.strip().split(':')
         paths: List[str] = []
-        for path in package_paths:
-            try:
-                all_packages = files.find(path, 'package.xml')
-            except dockerblade.exceptions.DockerBladeException:
-                logger.warning('unable to find directory in ROS_PACKAGE_PATH:'
-                               f' {path}')
-                continue
-            package_dirs = [os.path.dirname(p) for p in all_packages]
+        if files.exists('/opt/ros/dashing/'):
+            package_names = '/opt/ros/dashing/share/' \
+                            + 'ament_index/resource_index' \
+                            + '/packages'
+            all_packages = files.listdir(package_names)
+            package_dirs = []
+            for p in all_packages:
+                if files.exists('/opt/ros/dashing/share/' + p):
```

Comment:
For this particular implementation, this if condition is a tautological. Each `p` comes from `listdir`, so we know that the file definitely exists. (If there's a case where this isn't true, then there's a nasty bug lurking.)